quickwit/quickwit-search/src/jaeger_collector.rs

```diff
@@ -0,0 +1,226 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use std::collections::HashSet;
+use std::sync::Arc;
+
+use base64::prelude::{Engine, BASE64_STANDARD};
+use fnv::FnvHashMap;
+use itertools::Itertools;
+use serde::{Deserialize, Serialize};
+use tantivy::collector::{Collector, SegmentCollector};
+use tantivy::fastfield::{Column, MultiValuedFastFieldReader};
+use tantivy::{DateTime, DocId, InvertedIndexReader, Score, SegmentReader};
+
+type B64TraceId = String;
+type TraceId = Vec<u8>;
+type TraceIdTermOrd = u64;
+type SpanTimestamp = i64;
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct TraceIdSpanTimestamp {
+    pub trace_id: B64TraceId,
+    pub span_timestamp: SpanTimestamp,
+}
+
+impl TraceIdSpanTimestamp {
+    fn from_base64(trace_id: B64TraceId, span_timestamp: i64) -> Self {
+        Self {
+            trace_id,
+            span_timestamp,
+        }
+    }
+
+    fn from_bytes(trace_id: TraceId, span_timestamp: i64) -> Self {
+        Self {
+            trace_id: BASE64_STANDARD.encode(trace_id),
+            span_timestamp,
+        }
+    }
+}
+
+/// Finds the most recent trace ids among a set of matching spans. Multiple spans that belong to the
+/// same trace can be found in the document set. As a result, this problem is akin to finding the
+/// top k elements with duplicates
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct FindTraceIdsCollector {
+    num_traces: usize,
+    trace_id_field_name: String,
+    span_timestamp_field_name: String,
+}
+
+impl FindTraceIdsCollector {
```

Comment:
This PR implements a naive approach and focuses on plugging in the custom collector. I'll shortly follow up with a PR dedicated to improving the collector's performance.